### PR TITLE
GFORMS-949 Replace invisible characters from posted data with space

### PR DESCRIPTION
**Pull request by girishkamat**

### CODE REVIEW

File: app/uk/gov/hmrc/gform/controllers/helpers/InvisibleCharsStripper.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    * U+2008    Punctuation Space      &#8200;      " "
+    * U+2009    Thin Space             &#8201;      " "
+    * U+200A    Hair Space             &#8202;      " "
+    * U+200B    Zero-Width Space       &#8203;      "​"
+    * U+200C    Zero Width Non-Joiner  &#8204;      "‌"
+    * U+200D    Zero Width Joiner      &#8205;      "‍"
+    * U+200E    Left-To-Right Mark     &#8206;      "‎"
+    * U+200F    Right-To-Left Mark     &#8207;      "‏"
+    * U+202F    Narrow No-Break Space  &#8239;      " "
+    * U+FEFF    Byte order mark
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/controllers/helpers/InvisibleCharsStripper.scala:39]: How were set of these characters chosen? Why these and no others?

Comment (girishkamat) [reply app/uk/gov/hmrc/gform/controllers/helpers/InvisibleCharsStripper.scala:39]: I found these on stack overflow. These are the characters that represent empty and space i.e not visible to user. If there are more characters, we can update.

https://stackoverflow.com/questions/17978720/invisible-characters-ascii

Comment (girishkamat) [reply app/uk/gov/hmrc/gform/controllers/helpers/InvisibleCharsStripper.scala:39]: This probably has a more extensive list

https://gist.github.com/JamoCA/42c3be286185aff0476d5888f0a819ff

Comment (girishkamat) [reply app/uk/gov/hmrc/gform/controllers/helpers/InvisibleCharsStripper.scala:40]: Updated with `POP DIRECTIONAL FORMATTING` U+202C

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**